cli/azd/pkg/input/console.go

```diff
@@ -119,6 +121,9 @@ type AskerConsole struct {
 	// AskerConsole can be used as a singleton, hence, more than one component can invoke its methods at the same time.
 	// A method should lock this mutex if no other writing to he terminal should occur at the same time.
 	writeControlMutex sync.Mutex
+	// holds the last 2 bytes written by message or messageUX. This is used to detect when there is already an empty
+	// line (\n\n)
+	last2Byte [2]byte
```

Comment:
Part of me wonders if there is a way to have the console track this state in a more direct way, instead of requiring every place that writes text to also call `updateLastBytes`.  I wonder if it would be possible to do this by having some io.Writer that wraps the underlying writer and updates the state after it forwards the call to the underlying thing.  But perhaps that does not compose well with some of the external packages we are using...

Just a little nervous that we are going to forget to update the state in all the places as we evolve this code over time.

Reply:
100% agree here.
Even wrapping the io.Writer with some FooWriter, as you pointed out, someone could forget to use the `FooWriter` and directly write to stdOut.

Would you be fine with me creating a new issue to follow up on building the internal layer inside the console implementation to handle the blank line detection?

Reply:
issue: https://github.com/Azure/azure-dev/issues/2798

Reply:
> Would you be fine with me creating a new issue to follow up on building the internal layer inside the console implementation to handle the blank line detection?

Yup, SGTM (sounds good to me)